homeassistant/components/somfy/switch.py

```diff
@@ -26,9 +29,9 @@ def get_shutters():
 class SomfyCameraShutter(SomfyEntity, SwitchEntity):
     """Representation of a Somfy Camera Shutter device."""
 
-    def __init__(self, device, api):
+    def __init__(self, coordinator, device_id, api):
         """Initialize the Somfy device."""
-        super().__init__(device, api)
+        super().__init__(coordinator, device_id, api)
         self.shutter = CameraProtect(self.device, self.api)
```

Comment:
Since the coordinator entity has `should_poll` set to False, this switch entity will no longer recreate the shutter at the end of every service call. What impact will that have?

Reply:
Good call.  It looks like get position will be wrong for devices that aren't assumed state.  I'll fix it in a follow up.

Reply:
I just test the dev branch with your fix, and the cover position is well updated.

Reply:
Indeed, I just test it, and the update state is now broken.

Reply:
Fixed here https://github.com/home-assistant/core/pull/42462